src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -550,6 +550,26 @@ public void EmptyEnumStoreTest()
             Assert.True(enums.Count() > 0);
         }
 
+        [Fact]
+        public async void MaxRecursionDepthTest()
+        {
+            var config = new PowerFxConfig(null)
+            {
+                MaxCallDepth = 5
+            };
+            var recalcEngine = new RecalcEngine(config);
+            try
+            {
+                await recalcEngine.EvalAsync("Abs(Abs(Abs(Abs(Abs(Abs(1))))))", new System.Threading.CancellationToken());
+                Assert.True(false);
+            }
+            catch (MaxCallDepthException)
+            { 

```

Comment:
In the body of this set a boolean flag to true. For example callDepthCaught = true. Then assert that it is true later.